src/routes/transactions/mappers/common/human-description.mapper.ts

```diff
@@ -0,0 +1,115 @@
+import { Hex } from 'viem/src/types/misc';
+import { Inject, Injectable } from '@nestjs/common';
+import { decodeFunctionData, formatUnits, isHex } from 'viem';
+import { ITokenRepository } from '../../../../domain/tokens/token.repository.interface';
+import { TokenRepository } from '../../../../domain/tokens/token.repository';
+import { Token } from '../../../../domain/tokens/entities/token.entity';
+import { MAX_UINT256 } from '../../../../routes/transactions/constants';
+import {
+  ILoggingService,
+  LoggingService,
+} from '../../../../logging/logging.interface';
+import { SafeAppInfo } from '../../../transactions/entities/safe-app-info.entity';
+import { IHumanDescriptionRepository } from '../../../../domain/human-description/human-description.repository.interface';
+import { HumanDescriptionRepository } from '../../../../domain/human-description/human-description.repository';
+import {
+  HumanDescriptionFragment,
+  ValueType,
+} from '../../../../domain/human-description/entities/human-description.entity';
+
+@Injectable()
+export class HumanDescriptionMapper {
+  constructor(
+    @Inject(ITokenRepository) private readonly tokenRepository: TokenRepository,
+    @Inject(LoggingService) private readonly loggingService: ILoggingService,
+    @Inject(IHumanDescriptionRepository)
+    private readonly humanDescriptionRepository: HumanDescriptionRepository,
+  ) {}
+
+  async mapHumanDescription(
+    to: string | undefined,
+    data: string | null,
+    chainId: string,
+    safeAppInfo: SafeAppInfo | null,
+  ): Promise<string | null> {
+    if (!data || !isHex(data) || !to) return null;
+
+    const parsedDescriptions =
+      this.humanDescriptionRepository.getDescriptions();
+
+    const dataStart = data.slice(0, 10);
+    const sigHash = isHex(dataStart) ? dataStart : null;
+
+    if (!sigHash) return null;
+
+    const template = parsedDescriptions[sigHash];
```

Comment:
This can be undefined but is currently not caught because we don't have `"noUncheckedIndexedAccess": true` in the tsconfig. Activating that rule breaks a lot of other places too so I suggest to change it in a separate PR.